From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Thu, 24 Aug 2017 03:50:11 +0200
Subject: [PATCH] salt-master: Increase command_timeout to 15s

Default command timeout (10s) is sometimes too small for all minions
to return within the required timeslot. Increase it by 50%.

While at it, increase the timeout used for executing the "linux"
state against all nodes to 60s, just to be on the safe side.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 mcp/config/states/maas                                                  | 2 +-
 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/mcp/config/states/maas b/mcp/config/states/maas
index 12ef9ed..14f32ee 100755
--- a/mcp/config/states/maas
+++ b/mcp/config/states/maas
@@ -107,4 +107,4 @@ wait_for 10 "salt -C '* and not cfg01* and not mas01*' ssh.set_auth_key ${SUDO_U

 wait_for 10 "salt -C '* and not cfg01* and not mas01*' saltutil.sync_all"
 wait_for 10 "salt -C '* and not cfg01* and not mas01*' state.apply salt"
-wait_for 10 "salt -C '* and not cfg01* and not mas01*' state.apply linux,ntp"
+wait_for 10 "salt -C '* and not cfg01* and not mas01*' state.apply linux,ntp -t 60"
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml
index 7d95ebc..46526da 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml
@@ -37,6 +37,7 @@ parameters:
   salt:
     master:
       accept_policy: open_mode
+      command_timeout: 15
   reclass:
     storage:
       data_source:
